### PR TITLE
[core][App Router] Fix components from client component map not being code extracted

### PR DESCRIPTION
**Pull request by art-alexeyenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/content-sdk/blob/dev/CONTRIBUTING.md)
- [ ] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
Fixes the issue where component-map.client.ts was not considered for code extraction

## Testing Details
- [x] Unit Test Added
- [x] Manual Test/Other - stuff works (Image is client component)
image

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/core/src/tools/codegen/extract-files.ts [hunk 1]
```diff
@@ -128,7 +140,7 @@ function _extractFiles(args: ExtractFilesConfig = {}) {
         )
       );
     } catch (error) {
-      console.error(chalk.red('Error during code extraction:', error));
+      console.error(chalk.red('Error during code extraction:', error, error.stack));
```
Comment (illiakovalenko) [inline packages/core/src/tools/codegen/extract-files.ts:143]: Probably it's a leftover? We don't really need to show stack in logs

File: packages/core/src/tools/codegen/extract-files.test.ts [hunk 2]
```diff
@@ -276,6 +276,44 @@ describe('extract-files', () => {
           );
           expect(msgs).to.include(successMsg);
         });
+
+        xit('should call sendCode for each component path and package.json when client component map present', async () => {
```
Comment (illiakovalenko) [inline packages/core/src/tools/codegen/extract-files.test.ts]:
```suggestion
        it('should call sendCode for each component path and package.json when client component map present', async () => {
```

File: packages/core/src/tools/codegen/extract-files.test.ts
Comment (illiakovalenko) [inline packages/core/src/tools/codegen/extract-files.test.ts:1]: We need to add unit tests to cover main scenarios:
* New `clientComponentMapPath` parameter
* Absolute/relative paths
* An edge case when path is present but client map doesn't exist

**Review by illiakovalenko — APPROVED** [review]
Looks good! 👍
I’ve left some comments below - the PR is approved in advance so it’s not blocked, but please make sure to address the feedback
